clerk/clerk_options.go

```diff
@@ -0,0 +1,37 @@
+package clerk
+
+import (
+	"errors"
+	"net/http"
+)
+
+// ClientOption describes a functional parameter for the client constructor
+type ClientOption func(*client) error
+
+// WithHTTPClient allows the overriding of the http client
+func WithHTTPClient(httpClient *http.Client) ClientOption {
+	return func(c *client) error {
+		if httpClient == nil {
+			return errors.New("http client can't be nil")
+		}
+
+		c.client = httpClient
+		return nil
+	}
+}
+
+func WithBaseURL(rawURL string) ClientOption {
```

Comment:
:upside_down_face: Let's add a godoc comment here as well.

Reply:
Nice catch missed that, thanks